crates/models/src/journals.rs

```diff
@@ -51,24 +51,35 @@ pub struct S3BucketAndPrefix {
     #[validate]
     #[serde(default)]
     pub prefix: Option<Prefix>,
+
+    /// AWS region of the S3 bucket. Defaults to us-east-1 if unset, for backward compatibility with
+    /// pre-existing storage mappings that did not have this configuration and would only work if
+    /// the bucket was in us-east-1.
+    pub region: Option<String>,
 }
 
-impl S3BucketAndPrefix {
+impl S3StorageConfig {
     fn as_url(&self) -> url::Url {
         // These are validated when we validate storage mappings
         // to at least be legal characters in a URI
-        url::Url::parse(&format!(
+        let mut u = url::Url::parse(&format!(
             "s3://{}/{}",
             self.bucket,
             self.prefix.as_deref().unwrap_or("")
         ))
-        .expect("parsing as URL should never fail")
+        .expect("parsing as URL should never fail");
+
+        u.query_pairs_mut()
+            .append_pair("region", self.region.as_deref().unwrap_or("us-east-1"));
```

Comment:
I'm wondering if we can avoid hard coding `us-east-1` as the default. What happens if we leave it unspecified? Would that cause the default to be whatever is in the aws config file on the brokers? If so, then it kinda seems preferable.

Reply:
Yep it would use the default from the AWS config of the brokers. I don't have a compelling reason not to just do that, so I've made the update.